PowerShell/JumpCloud Module/Private/Users/Delete-JcUser.ps1

```diff
@@ -0,0 +1,67 @@
+function Delete-JCUser {
+    # UserId and cascade_manager are required
+    param (
+        [Parameter(Mandatory = $true, ValueFromPipelineByPropertyName = $true, HelpMessage = 'The JumpCloud User ID.')][ValidateNotNullOrEmpty()][System.String]$Id
+        , [Parameter(Mandatory = $false, ValueFromPipelineByPropertyName = $true, HelpMessage = 'The JumpCloud cascade manager ID.')][System.String]$managerId,
+        # headers are required
+        [Parameter(Mandatory = $true, ValueFromPipelineByPropertyName = $true, HelpMessage = 'The JumpCloud API headers.')][ValidateNotNullOrEmpty()][System.Collections.Hashtable]$Headers,
+        [Parameter(Mandatory = $false, ValueFromPipelineByPropertyName = $true, HelpMessage = 'Force the deletion of the user.')][System.Management.Automation.SwitchParameter]$force,
+        # Parameter for $UserHash hashtable
+        [Parameter(Mandatory = $true, ValueFromPipelineByPropertyName = $true, HelpMessage = 'The JumpCloud User Hash.')][System.Collections.Hashtable]$UserHash
+    )
+    begin {
+        $Username = $UserHash.GetEnumerator().Where({ $_.Name -contains ($Id) }) | Select-Object -ExpandProperty Value | Select-Object -ExpandProperty username
+    }
+    process {
+        try {
+            if ($managerId) {
+                $URI = "$JCUrlBasePath/api/systemusers/$($Id)?cascade_manager=$($managerId)"
+                # Get the manager's username from $Userhash
+                $managerUsername = $UserHash.GetEnumerator().Where({ $_.Name -contains ($managerId) }) | Select-Object -ExpandProperty Value | Select-Object -ExpandProperty username
+                Write-Host "Deleting user: $($Username) and cascading managed users to manager: $($ManagerUsername)" -ForegroundColor Yellow
+                $prompt = Read-Host "Are you sure you wish to delete the user: $($Username)? (Y/N)"
+                while ($prompt -ne 'Y' -and $prompt -ne 'N') {
```

Comment:
Just per the other comments, I'm confused why we have a read host in `Remove-JCUser` and are again prompting within the private function. This should be reworked IMO to only ask for input in one place and have actions depending on that behavior.

Reply:
Moved the functionality to remove-JCUser